### ISSUE TITLE
Documentation for configuration properties for different environments

### ISSUE THREAD

**Issue by ajoslin103** [react-boilerplate/react-boilerplate#2820]
**Is your feature request related to a problem? Please describe.**
I need to use some config params that will be in Heroku environment variables when I deploy in production.

I read all of the project issues I could find on the subject and am frustrated that can't figure this out even with all the hints from people who seem to have made this work

**Describe the solution you'd like**
A cookbook solution in the Heroku deployment docs, something I can paste in and get on with trying to figure out everything else that stumps me.

**Describe alternatives you've considered**
I tried to use the dotenv solutions, including the webpack file mods -- starting with the suggestions and modifying them so that they would compile -- but the result in my code is always `undefined`

**Additional context**
I guess context is what I want...

Actually, I really hope that the answer is here already, that I just missed the capability that is already resident in the boilerplate

Thanks for react-boilerplate, I am finding it very helpful!

**Comment by julienben** [comment]
Can you please describe more precisely what you are trying to do? I'm struggling to understand. 

What do you mean by "Heroku environment variables"? "Config params"? 

You tried Webpack's EnvironmentPlugin?

**Comment by ajoslin103** [comment]
Apologies for the confusing request, I do not know how some of these things work behind the scenes

Heroku, in their Settings tab, has Config Vars -- which I understand to be environment variables 

If I set an environment variable in development using some process explained here, then I would like to NOT see that value when I get to production. Instead having that same environment variable supplied by the Heroku Config Vars.

I just tried the Webpack EnvironmentPlugin -- adding this to my webpack.dev.babel.js file at the top of the plugins section

```
  // Add development plugins
  plugins: [
    new webpack.EnvironmentPlugin({
      'CORS_PROXY':
      'https://cors-anywhere.herokuapp.com/',
    }),
    new webpack.HotModuleReplacementPlugin(), // Tell webpack we want hot reloading
    [etc...]
```
Then I tried to use it in a container:

`const process = require('process');
`
```
console.log(process.env); 
// this outputs {} -- I was hoping to see { CORS_PROXY: 'https://cors-anywhere.herokuapp.com/' }
```

I do apologize for not understanding how these things work, I see that the webpack.EnvironmentPlugin has been around for quite a while - so it's not really a react-boilerplate issue.  Therefore I really appreciate your time on this, and I would submit a pull request for the HHG docs explaining what I've learned for the next less-than-hoopy frood.

**Comment by ajoslin103** [comment]
Ah! It's working now -- odd, since I've not re-visited this issue in a couple of days.  But irrelevant, since it's working now.  

All that remains is to understand it.

Webpack is quietly replacing my usages of process.env.CORS_PROXY with the supplied value during compilation. 

So when I ask what's in process.env as an object -- nothing will be there because I gave webpack nothing to use in such a replacement.

I am so used to writing code that will be executed, with bugs and all -- but webpack compiles. And because of that this EnvironmentPlugin is the equivalent of the old compiler definitions.  It has been a long time, and I forgot about that way of thinking. Webpack just works, and no matter how many times I watched it 'compiling...' I wasn't thinking about it as a compiler.  Ha!

When I deploy to Heroku it will compile there, and any Config vars will (or may not) be available so for best effect I need to code defensively when using EnvironmentPlugin -- the value may be undefined.

Thanks for the pointer to EnvironmentPlugin!!
